File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/SavedRequestCookieConfigurationProperties.java
package uk.gov.justice.digital.hmpps.oauth2server.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

import java.time.Duration;

@Configuration
@ConfigurationProperties(prefix = "saved-request.cookie")
@Data
public class SavedRequestCookieConfigurationProperties {
    private String name;
    private Duration expiryTime;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/SavedRequestCookieConfigurationProperties.kt
package uk.gov.justice.digital.hmpps.oauth2server.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import java.time.Duration

@ConstructorBinding
@ConfigurationProperties(prefix = "saved-request.cookie")
data class SavedRequestCookieConfigurationProperties(val name: String, val expiryTime: Duration)
